### PR TITLE
Confirm reservation

### PR DESCRIPTION
**Pull request by Pochetes**
## Confirmation Page
- when users click on `Reserve Now` button in `ListingDetail`, a reservation of status `DRAFT` is created and are led to a confirmation page
- a summary of your reservation, pool listing details and transaction details are shown in the confirmation page
- calculated subtotal, service fees, taxes and total, referenced from the backend
- when users confirm reservation, their reservation gets updated to a status of `CONFIRMED`
- globalized reservations to be able to use everywhere in a listing and its constituents, avoiding prop-drilling
- when fetching for reservations from a listing, the sql query now specifies reservations with a status of either `DRAFT` or `CONFIRMED`

## Demo
when the user inputs their reservation dates and times, the backend creates a new reservation entity with the status of `DRAFT`
Screen Shot 2022-08-03 at 10 38 26 AM

They are then navigated to a confirmation page, where they review their details. Once they press the confirmation button, the same reservation entity's status is updated to `CONFIRMED`
Screen Shot 2022-08-03 at 10 40 37 AM

In the future, I want to handle cases when the user backs out of the reservation, which would update the status to `ARCHIVED`. I learned that nothing should ever be deleted unless it's absolutely necessary. Instead, it should be stored away for recovery purposes. Another important case would if the user spends more than 10 minutes confirming the reservation using a timeout handler. This will also trigger a mutation in the reservation's status as `ARCHIVED`

### CODE REVIEW

File: swimster-api/routes/reservations.js [hunk 1]
```diff
@@ -55,4 +55,15 @@ router.post('/listings/:listingId', requiresAuth, async (req, res, next) => {
     }
 })
 
+router.put('/listings/:listingId', requiresAuth, async (req, res, next) => {
+    try {
+        // changes status of DRAFT reservation to CONFIRMED
```
Comment (roeshanigos) [inline swimster-api/routes/reservations.js:60]: where exactly does it set it to DRAFT?

Comment (roeshanigos) [reply swimster-api/routes/reservations.js:60]: talked to Roberto offline - the schema sets the value of status when a reservation is created is created (when they hit Reserve button)

**Review by roeshanigos — COMMENTED** [review]
to be nitpicky `1 hours` should be `1 hour `Screen Shot 2022-08-04 at 2 34 02 PM

File: swimster-ui/src/components/ReservationCard/ReservationCard.jsx [hunk 2]
```diff
@@ -68,15 +74,13 @@ const ReservationCard = ({ price, register, setValue, isSubmitProcessing, handle
...
                             placeholder='Guests'
                             className={style.reservationGuestsInput}
-                            {...register("reservationGuests")}
+                            {...register("reservationGuests", {
+                                onChange: (e) => setGuests(e.target.value)
+                            })}
                         />
                     </div>
                 </div>
-                {isSubmitProcessing ? (
```
Comment (roeshanigos) [inline swimster-ui/src/components/ReservationCard/ReservationCard.jsx:75]: here

File: swimster-ui/src/hooks/useReservationCard.js [hunk 3]
```diff
@@ -1,127 +1,89 @@
...
+        }
+
+        const { data, error } = await apiClient.createReservation(JSON.stringify(formattedFormData), listingId)
+        if (error) setError(error)
+        if (data?.reservation) {
+            setReservation(data.reservation)
+            navigate('confirm')
+        }
+
+        setIsSubmitProcessing(false)
```
Comment (roeshanigos) [inline swimster-ui/src/hooks/useReservationCard.js:71]: it seems that you arent using IsSubmitProcessing anymore? is there a need for it now?

File: swimster-api/routes/reservations.js (hunk 1 above)
Comment (Pochetes) [reply swimster-api/routes/reservations.js:60]: done!

File: swimster-ui/src/hooks/useReservationCard.js (hunk 3 above)
Comment (Pochetes) [reply swimster-ui/src/hooks/useReservationCard.js:71]: I will use loading to navigate to the confirmation page. There will be a PR for loading very soon.

**Review by roeshanigos — APPROVED** [review]

File: swimster-api/models/Reservation.js [hunk 4]
```diff
@@ -192,6 +194,22 @@ class Reservation {
         const reservations = result.rows
         return reservations
     }
+
+    static async confirmReservation(reservationId) {
```
Comment (roeshanigos) [inline swimster-api/models/Reservation.js:198]: for reference this is where reservation status changes to confrim

**Pull request merged**